[DELAY MERGE] TNL-6145: Upgrade ES to 1.x for notes

robrap opened the pull request:
## [TNL-6145](https://openedx.atlassian.net/browse/TNL-6145)

### Description

This is to upgrade ES to 1.5 for notes.  Mostly, it is about removing support for ES 0.9.

A version of this PR (https://github.com/edx/edx-notes-api/pull/34) already had thumbs, but this also makes haystack a little more up to date and drops ES 1.4 testing.

This will be a slightly longer lasting branch until we are ready for deployment of ES 1.5 everywhere.

### Sandbox
- Not available.

### Testing
- [X] Unit, integration, acceptance tests as appropriate
- [x] Performance

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @maxrothman 
- [x] Code review: @dianakhuang 

FYI: @clintonb 

### Post-review
- [x] Rebase and squash commits [DO NOT MERGE]

clintonb commented on README.rst:
```diff
@@ -10,13 +10,12 @@ This is a backend store for edX Student Notes.
 ---------------
 
-1. You'll need an `ElasticSearch <http://elasticsearch.org>`__ installed.
+1. Install `ElasticSearch <http://elasticsearch.org>`__ 1.x.y.
```
The final 1.x.y version is 1.5.2. Just link to it directly: https://www.elastic.co/blog/elasticsearch-1-5-2-and-1-4-5-released

robrap replied:
Sure, I can update.  Also, this repo is soon to go through the Docker changes in the devops backlog as part of the OS upgrade work.

robrap commented:
FYI: @maxrothman @dianakhuang: Notes was dockerized for testing since this was created so I just rebased and fixed and squashed.  Hopefully all is well with the tests.
